src/main/java/io/cryostat/net/web/http/api/beta/RecordingDeleteHandler.java

```diff
@@ -0,0 +1,123 @@
+ *
+ * without restriction, including without limitation the rights to copy, create
+ * derivative works of, display, perform, and distribute the Software and make,
+ * use, sell, offer for sale, import, export, have made, and have sold the
+ * Software and the Larger Work(s), and to sublicense the foregoing rights on
+ * either these or other terms.
+ *
+ * This license is subject to the following condition:
+ * The above copyright notice and either this complete permission notice or at
+ * a minimum a reference to the UPL must be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package io.cryostat.net.web.http.api.beta;
+
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+
+import javax.inject.Inject;
+
+import io.cryostat.net.AuthManager;
+import io.cryostat.net.security.ResourceAction;
+import io.cryostat.net.web.http.HttpMimeType;
+import io.cryostat.net.web.http.api.ApiVersion;
+import io.cryostat.net.web.http.api.v2.AbstractV2RequestHandler;
+import io.cryostat.net.web.http.api.v2.ApiException;
+import io.cryostat.net.web.http.api.v2.IntermediateResponse;
+import io.cryostat.net.web.http.api.v2.RequestParameters;
+import io.cryostat.recordings.RecordingArchiveHelper;
+import io.cryostat.recordings.RecordingNotFoundException;
+
+import com.google.gson.Gson;
+import io.vertx.core.http.HttpMethod;
+
+public class RecordingDeleteHandler extends AbstractV2RequestHandler<Void> {
+
+    static final String PATH = "recordings/:sourceTarget/:recordingName";
+
+    private final RecordingArchiveHelper recordingArchiveHelper;
+
+    @Inject
+    RecordingDeleteHandler(
+            AuthManager auth, Gson gson, RecordingArchiveHelper recordingArchiveHelper) {
+        super(auth, gson);
+        this.recordingArchiveHelper = recordingArchiveHelper;
+    }
+
+    @Override
+    public boolean requiresAuthentication() {
+        return true;
+    }
+
+    @Override
+    public ApiVersion apiVersion() {
+        return ApiVersion.BETA;
+    }
+
+    @Override
+    public HttpMethod httpMethod() {
+        return HttpMethod.DELETE;
+    }
+
+    @Override
+    public Set<ResourceAction> resourceActions() {
+        return EnumSet.of(ResourceAction.DELETE_RECORDING);
+    }
+
+    @Override
+    public String path() {
+        return basePath() + PATH;
+    }
+
+    @Override
+    public List<HttpMimeType> produces() {
+        return List.of(HttpMimeType.PLAINTEXT);
```

Comment:
For all these new V2 handlers the `produces` list should be `List.of(HttpMimeType.JSON)`, I think, except for the ones that produce a report (`HTML`) or recording (`OCTET_STREAM`). The response format here is the standard API V2+ JSON response, not only plain text. This is a bit of a change from the older behaviour where the V2 API mimetype only reflected the type of the data contained within the result of the JSON response message. Since #1072 and the content negotiation implementation it now reflects the type of data of the response as a whole, not just one subfield within.